apps/mobile/src/components/Feed/Socialize/AdmireButton.tsx

```diff
@@ -0,0 +1,218 @@
+
+        pageInfo?.setValue(((pageInfo?.getValue('total') as number) ?? 1) - 1, 'total');
+
+        if (response.removeAdmire.admireID) {
+          const relayId = `Admire:${response.removeAdmire.admireID}`;
+
+          store.delete(relayId);
+        }
+      }
+    };
+
+    try {
+      const response = await removeAdmire({
+        updater,
+        optimisticUpdater: updater,
+        optimisticResponse: {
+          removeAdmire: {
+            __typename: 'RemoveAdmirePayload',
+            admireID: event.viewerAdmire.dbid,
+          },
+        },
+        variables: {
+          admireId: event.viewerAdmire.dbid,
+        },
+      });
+
+      if (
+        response.removeAdmire?.__typename !== 'RemoveAdmirePayload' &&
+        // We can silently fail if the post was already not admired
+        response.removeAdmire?.__typename !== 'ErrAdmireNotFound'
+      ) {
+        // TODO: handle or track error whenever we setup tracker
+      }
+    } catch (error) {
+      // TODO: handle or track error whenever we setup tracker
+    }
+  }, [event.viewerAdmire?.dbid, interactionsConnection, removeAdmire]);
+
+  const handleAdmire = useCallback(async () => {
+    if (query.viewer?.__typename !== 'Viewer') {
+      return;
+    }
+    const updater: SelectorStoreUpdater<AdmireButtonMutation['response']> = (store, response) => {
+      if (response?.admireFeedEvent?.__typename === 'AdmireFeedEventPayload') {
+        const pageInfo = store.get(interactionsConnection)?.getLinkedRecord('pageInfo');
+
+        pageInfo?.setValue(((pageInfo?.getValue('total') as number) ?? 0) + 1, 'total');
+      }
+    };
+
+    try {
+      const optimisticAdmireId = Math.random().toString();
+      const response = await admire({
+        updater,
+        optimisticUpdater: updater,
+        optimisticResponse: {
+          admireFeedEvent: {
+            __typename: 'AdmireFeedEventPayload',
+            feedEvent: {
+              id: event.id,
+              viewerAdmire: {
+                __typename: 'Admire',
+                id: `Admire:${optimisticAdmireId}`,
+                dbid: optimisticAdmireId,
+                creationTime: new Date().toISOString(),
+                admirer: {
+                  id: query.viewer?.user?.id ?? 'unknown',
+                  dbid: query.viewer?.user?.dbid ?? 'unknown',
+                  username: query.viewer?.user?.username ?? null,
+                },
+              },
+            },
+          },
+        },
+        variables: {
+          eventId: event.dbid,
+          connections: [interactionsConnection, notesModalConnection],
+        },
+      });
+
+      if (
+        response.admireFeedEvent?.__typename !== 'AdmireFeedEventPayload' &&
+        // We can silently fail if the post was already admired
+        response.admireFeedEvent?.__typename !== 'ErrAdmireAlreadyExists'
+      ) {
+        // TODO: Handle this error
+      }
+    } catch (error) {
+      // TODO: Handle this error
```

Comment:
Gonna add this later whenever we have error tracker just like in web

Reply:
We do have an error tracker. `useReportError` works on mobile. You can see usage in `EnterEmailScreen` or `useLogout`